xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/createinline.vm

```diff
@@ -67,6 +67,17 @@
       )
   </div>
 #end
+## check if the exception is about the document having a path too long
+#if("$!exception" != '' && $exception.code == 11017)
```

Comment:
Another solution would be to let the query execute and we capture the JDBC exception if there's an error instead. And possibly translate it into a user friendly message. This would capture not only size issues but also any other problem, no?

Reply:
> This would capture not only size issues but also any other problem, no?

yes it would, but the problem here is about "And possibly translate it into a user friendly message.": it's not necessarily always easy. It's actually easier to check the problem before it happens. 

Besides it's safer to check the problem before the exception, especially when several documents are involved, like in a move of a hierarchy. So this PR is a first step for adding those checks.